src/ui/public/state_management/__tests__/monitor_changes.js

```diff
@@ -0,0 +1,107 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+import monitorStateChanges from 'ui/state_management/monitor_changes';
+import SimpleEmitter from 'ui/utils/SimpleEmitter';
+
+describe('monitorStateChanges', function () {
+  const noop = () => {};
+  const eventTypes = [
+    'save_with_changes',
+    'reset_with_changes',
+    'fetch_with_changes',
+  ];
+  let mockState;
+
+  beforeEach(() => {
+    mockState = new SimpleEmitter();
+  });
+
+  describe('sanity checks', function () {
```

Comment:
Is there a way to rename this block so that the log would have a more human-readable message? E.g. right now it will be:

```
monitorStateChanges sanity checks should throw if not given a handler function
```

But the concept of sanity checks doesn't exist in the interface.  Maybe if we just name `sanity checks` to `should throw errors`, then it will read more clearly:

```javascript
monitorStateChanges should throw errors if not given a handler function
```